lib/expression/v2/callbacks.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Expression.V2.Callbacks do
+  @moduledoc """
+  Callback functions to be used in Expressions.
+
+  This is the same idea as `Expression.Callbacks.Standard` but 
+  it's in a rough shape, mostly to just prove that this all works.
```

Comment:
This made me think about what would be the plan to "migrate" from v1 to v2. Also would we have to change anything for our custom callbacks?

Reply:
We'll need to copy over the stuff from the existing callbacks and rework them to use the new format. I'm still wanting a way to have each function describe by its own name so we can document them easily using the tools we have instead of having a pile of `callback/3` functions that pattern match.

We'll get to that, right now I'm mostly trying to make sure that the implementation itself is good and then see what we can do to make it easier to document & implement the callbacks.

Reply:
This was handled in #102 and has been merged into this PR